core/shared/src/test/scala/fs2/StreamInterruptSuite.scala

```diff
@@ -369,4 +369,15 @@ class StreamInterruptSuite extends Fs2Suite {
     val interrupt = IO.sleep(250.millis)
     IO.race(compileWithSync(s).drain, interrupt).map(_.isRight).assert
   }
+
+  test("issue #2842 - no interrupt scope") {
+    val s = for {
+      local <- Stream.eval(IOLocal(List.empty[Int]))
+      _ <- Stream.eval(local.update(1 :: _)).noInterruptScope
+      _ <- Stream.eval(local.update(2 :: _)) // this one will be lost on a forked fiber
+      _ <- Stream.eval(local.update(3 :: _)).noInterruptScope
+      result <- Stream.eval(local.get)
+    } yield result
+    s.interruptScope.compile.lastOrError.assertEquals(List(3, 1))
+  }
```

Comment:
This is what I was concerned about—that the `noInterruptScope` would leak to other portions of the `Stream`. This test currently fails, because `Stream.eval(local.update(2 :: _))` is not interruptible.